Tests/Unit/inc/classes/ImagifySettings/updateSiteOptionOnNetwork.php

```diff
@@ -0,0 +1,113 @@
+<?php
+namespace Imagify\Tests\Unit\inc\classes\ImagifySettings;
+
+use Brain\Monkey\Filters;
+use Brain\Monkey\Functions;
+use Imagify_Settings;
+use Imagify\Tests\Unit\TestCase;
+
+/**
+ * @covers Imagify_Settings::update_site_option_on_network
+ *
+ * @uses   imagify_check_nonce()
+ * @uses   imagify_die()
+ * @uses   imagify_maybe_redirect()
+ *
+ * @group  ImagifySettings
+ */
+class Test_UpdateSiteOptionOnNetwork extends TestCase {
+
+	public function setUp() {
+		parent::setUp();
+
+		if ( ! defined( 'IMAGIFY_SLUG' ) ) {
+			define( 'IMAGIFY_SLUG', 'imagify' );
+		}
+
+		Functions\when( 'imagify_is_active_for_network' )->justReturn( true );
+	}
+
+	/**
+	 * @dataProvider configTestData
+	 */
+	public function testShouldUpdateNetworkSettings( $config, $expected ) {
+		$_POST['option_page'] = $config['option_page'];
```

Comment:
Here too you'll want to reset between tests on tearDown

Reply:
Done